docs/api.md

```diff
@@ -34,3 +34,139 @@ These are the API group versions which are curently available when using Antrea.
+
+For the v1.0 release, we undertook to rename all Antrea API to use the
+`antrea.io` suffix instead of the `antrea.tanzu.vmware.com` suffix. For more
+information about the motivations behind this undertaking, please refer to
+[Github issue #1715](https://github.com/vmware-tanzu/antrea/issues/1715).
+
+As part of this renaming, and to avoid profileration of API groups, we have
+decided to group all the Custom Resource Definitions (CRDs) defined by Antrea in
+a single API group: `crd.antrea.io`. In the future, resources within that group
+will be versioned individually as needed, and this will be reflected in the
+table above.
+
+To avoid disruptions to existing Antrea users, our requirements for this
+renaming process were as follows:
+
+1. As per our [upgrade
+   policy](versioning.md#antrea-upgrade-and-supported-version-skew), older
+   Agents need to be able to communicate with a new upgraded Controller, using
+   the old `controlplane.antrea.tanzu.vmware.com` API. Once both the Controller
+   and the Agent are upgraded, they communicate using `controlplane.antrea.io`.
+2. API Services can be accessed using either API version.
+3. After upgrade, Custom Resources can be managed using either API
+   version. Resources created using the old API (before or after upgrade) can be
+   accessed using the new API (or the old one).
+4. For each resource in each API group, the new resource type should be
+   backward-compatible with the old resource type, and, whenever possible,
+   forward-compatible. This simplifies the upgrade of existing client
+   applications which leverage the Antrea API. These applications can be easily
+   upgraded to use the new API version, with no change to the business
+   logic. Custom Resources created before upgrading the application can be
+   accessed through the new API with no loss of information.
+
+To achieve our 3rd goal, we introduced a new Kubernetes controller in the Antrea
+Controller, in charge of mirroring "old" Custom Resources (created using the
+`*.antrea.tanzu.vmware.com` API groups) to the new (`*.antrea.io`) API. This new
+mirroring controller is enabled by default, but can be disabled by setting
+`legacyCRDMirroring` to `false` in the `antrea-controller` configuration
+options. Thanks to this controller, the Antrea components (Agent and Controller)
+only need to watch Custom Resources created with the new API group. If any
+client still uses the old (or "legacy") API groups, these Custom Resources will
+be mirrored to the new API group and handled as expected.
+
+The mirroring controller behaves as follows:
+
+* If a Custom Resource is created with the legacy API, it will create a new
+  Custom Resource with the same `Spec` and `Labels` as the legacy one.
+* Any update to the `Spec` and / or `Labels` of the legacy Custom Resource will
+  be reflected identically in the new Custom Resource.
+* Any update to the `Status` of the new mirrored Custom Resource will be
```

Comment:
This may be a bit redundant and verbose. But I think it could be more accurate.

Any update to the `Status` of the new mirrored Custom Resource will be ->
Any update to the `Status` of the new mirrored Custom Resource (If the Custom Resource has `Status`) will be

Reply:
No problem, I can add that